src/server/plugins/engine/plugin.ts

```diff
@@ -192,63 +205,118 @@ export const plugin = {
       const newValues = getValidStateFromQueryParameters(
         prePopFields,
         query,
         state
       )
+
       await cacheService.mergeState(request, newValues)
+
       return h.continue
     }
 
+    const loadFormPreHandler = server.app.model
+      ? async (request: Request, h: ResponseToolkit) =>
+          (request.app.model = server.app.model)
+      : async (request: Request, h: ResponseToolkit) => {
+          const { params } = request
+          const { slug } = params
+          /**
+           * @todo Determine from path params
+           */
+          const formState = 'draft'
+
+          // Get the form metadata using the `slug` param
+          const metadata = await getFormMetadata(slug)
+
+          const { id, [formState]: state } = metadata
+
+          // Check the metadata supports the requested state
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          if (!state) {
+            return Boom.notFound(
+              `No '${formState}' state for form metadata ${id}`
+            )
+          }
+
+          const key = `${id}_${formState}`
+          let item = itemCache.get(key)
+
+          if (!item || !isEqual(item.updatedAt, state.updatedAt)) {
+            server.logger.info(`Getting form definition ${id} (${slug})`)
+
+            // Get the form definition using the `id` from the metadata
+            const definition = await getFormDefinition(id, formState)
+
+            // Generate the form model and add it to the item cache
+            server.logger.info(
+              `Building model for form definition ${id} (${slug})`
+            )
+            const model = new FormModel(definition, {
+              basePath: slug,
+              ...modelOptions
+            })
+
+            // Create new cache item
+            item = { model, updatedAt: state.updatedAt }
+            itemCache.set(key, item)
+          }
+
+          // Assign the model to the request data
+          // for use in the downstream handler
+          request.app.model = item.model
+
+          return h.continue
+        }
+
     server.route({
       method: 'get',
-      path: '/{id}',
+      path: '/{slug}',
+      handler: (request: Request, h: ResponseToolkit) => {
+        const { slug } = request.params
+        const model = request.app.model
+
+        return getStartPageRedirect(request, h, slug, model)
+      },
       options: {
         pre: [
+          {
+            method: loadFormPreHandler
+          },
           {
             method: queryParamPreHandler
           }
-        ]
-      },
-      handler: (request: Request, h: ResponseToolkit) => {
-        const { id } = request.params
-        const model = forms[id]
-        if (model) {
-          return getStartPageRedirect(request, h, id, model)
+        ],
+        validate: {
+          params: Joi.object().keys({
+            slug: Joi.string().required()
+          })
```

Comment:
One for `formBySlugSchema` now we export it?

Reply:
I'll use `slugSchema` from `@defra/forms-model` for now. `formBySlugSchema` lives in `forms-manager` at the moment (but should probably be promoted to `@defra/forms-model` to allow for reuse).

Reply:
Resolved with 002344074e0211dd1ff9b97fc07627d77adbb45c